Detectors/EMCAL/calibration/include/EMCALCalibration/EMCALTimeCalibData.h

```diff
@@ -0,0 +1,113 @@
+
+#include "DetectorsCalibration/TimeSlotCalibration.h"
+#include "DetectorsCalibration/TimeSlot.h"
+#include "DataFormatsEMCAL/Cell.h"
+#include "EMCALBase/Geometry.h"
+#include "CCDB/CcdbObjectInfo.h"
+
+#include "Framework/Logger.h"
+#include "CommonUtils/MemFileHelper.h"
+#include "CCDB/CcdbApi.h"
+#include "DetectorsCalibration/Utils.h"
+#include <boost/histogram.hpp>
+#include <boost/histogram/ostream.hpp>
+#include <boost/format.hpp>
+
+#include <array>
+#include <boost/histogram.hpp>
+namespace o2
+{
+namespace emcal
+{
+
+// holds the time calibration parameters
+class TimeCalibParams
+{
+  std::array<float, 17776> arr;
+};
+
+// class containing the initialization parameters for histograms (time bins/range etc.)
+struct TimeCalibInitParams {
+ public:
+  unsigned int mTimeBins = 1500;
+  std::array<float, 2> mTimeRange = {-500., 1000.}; // time range in ns
+  unsigned int mEnergyBins = 5000;
+  std::array<float, 2> mEnergyRange = {0., 50.}; // energy range in GeV
+};
+
+class EMCALTimeCalibData
+{
+ public:
+  using Cells = o2::emcal::Cell;
+  using boostHisto = boost::histogram::histogram<std::tuple<boost::histogram::axis::regular<double, boost::use_default, boost::use_default, boost::use_default>, boost::histogram::axis::integer<>>, boost::histogram::unlimited_storage<std::allocator<char>>>;
+
+  o2::emcal::Geometry* mGeometry = o2::emcal::Geometry::GetInstanceFromRunNumber(300000);
+  int NCELLS = mGeometry->GetNCells();
+
+  EMCALTimeCalibData(TimeCalibInitParams par)
+  {
+    // boost histogram with amplitude vs. cell ID, specify the range and binning of the amplitude axis
+    mTimeHisto = boost::histogram::make_histogram(boost::histogram::axis::regular<>(par.mTimeBins, par.mTimeRange.at(0), par.mTimeRange.at(1), "t (ns)"), boost::histogram::axis::integer<>(0, NCELLS, "CELL ID"));
+  }
+
+  ~EMCALTimeCalibData() = default;
+
+  /// \brief Fill the container with the cell ID and amplitude and time information.
+  void fill(const gsl::span<const o2::emcal::Cell> data);
+
+  /// \brief Merge the data of two slots.
+  void merge(const EMCALTimeCalibData* prev);
+
+  /// \brief Check if enough data for calibration has been accumulated
+  bool hasEnoughData() const;
+
+  /// \brief Print a useful message about the container.
+  void print();
+
+  /// \brief Get number of events currently available for calibration
+  int getNEvents() const { return mEvents; }
+
+  /// \brief Get current histogram
+  boostHisto& getHisto() { return mTimeHisto; }
+  const boostHisto& getHisto() const { return mTimeHisto; }
+
+  /// \brief Set number of events available for calibration
+  void setNEvents(int ne) { mEvents = ne; }
+
+  void PrintStream(std::ostream& stream) const;
+
+  /// \brief Actual function where calibration is done. Has to be called in has enough data when enough data is there
+  o2::emcal::TimeCalibParams process(); // how should the return type be be called? Or dont I have to return something and directly push to CCDB in process?
```

Comment:
Should return o2::emcal::TimeCalibrationParams (Detectors/EMCAL/calib - library with EMCAL CCDB content)

In addition a dummy implementation for the function is missing, without there will be a linker error.

Reply:
added a dummy implementation and added the correct return type